src/bin/miri.rs

```diff
@@ -223,6 +235,9 @@ impl rustc_driver::Callbacks for MiriCompilerCalls {
             if num_failed > 0 {
                 eprintln!("{num_failed}/{total} SEEDS FAILED", total = many_seeds.seeds.count());
             }
+            // drop the tracing guard before exiting, so tracing calls are flushed correctly
+            #[allow(clippy::drop_non_drop)]
```

Comment:
You can probably avoid the lint by adding `impl Drop for TracingGuard` to make sure it is always a drop-type.